### PR TITLE
Use UI toolkit for variable table

### PR DESCRIPTION
**Pull request by kentarolim10**
# Description
- Uses jupyter-ui-toolkit for the table styling

# Old styling

image

Pr for #277

### CODE REVIEW

File: src/variableinspector.ts [hunk 1]
```diff
@@ -8,12 +8,25 @@ import { DockLayout, Widget } from '@lumino/widgets';
...
+  DataGridRow,
+  DataGridCell,
+  allComponents,
+  provideJupyterDesignSystem,
+  Select,
+  Option,
+  Search,
+  Button
+} from '@jupyter/web-components';
+provideJupyterDesignSystem().register(allComponents);
```
Comment (fcollonval) [inline src/variableinspector.ts:22]: You should import the `jp...` variant of the components.

Then could import as fine grain the proper components to optimize what is loaded:

```suggestion
provideJupyterDesignSystem().register(jpDataGrid(), jpDataGridRow(), ...);
```

File: src/variableinspector.ts [hunk 2]
```diff
@@ -356,28 +414,22 @@ namespace Private {
...
 
+  export function createCellTemplate(table: WebDataGrid): HTMLTemplateElement {
+    const template = document.createElement('template');
+    const container = document.createElement('div');
+    container.innerText = 'testing';
+    template.appendChild(container);
+    table.appendChild(template);
+    return template;
+  }
+
```
Comment (fcollonval) [inline src/variableinspector.ts:432]: Dead code?

```suggestion
```

File: style/base.css [hunk 3]
```diff
@@ -52,6 +52,7 @@
 
 .filter-search-container {
   display: flex;
+  align-items: center;
   padding: 0 1rem;
 }
```
Comment (fcollonval) [inline style/base.css:58]: The rule `.filter-type` should be removed

File: src/variableinspector.ts [hunk 4]
```diff
@@ -250,48 +291,62 @@ export class VariableInspectorPanel
...
-      cell = row.insertCell(1);
+      cell = document.createElement('jp-data-grid-cell') as DataGridCell;
       if (item.isMatrix) {
         cell.title = 'View Contents';
         cell.className = 'jp-VarInspector-inspectButton';
         const ico = searchIcon.element();
+        ico.className = 'icon-button';
         ico.onclick = (ev: MouseEvent): any => {
-          console.log('Click on ' + name);
+          console.log('Click on ' + item.varName);
```
Comment (fcollonval) [inline src/variableinspector.ts:314]: Let's remove this

```suggestion
```

File: src/variableinspector.ts [hunk 5]
```diff
@@ -250,48 +291,62 @@ export class VariableInspectorPanel
...
       cell.append(ico);
+      row.appendChild(cell);
 
       // Add onclick event for inspection
-      cell = row.insertCell(1);
+      cell = document.createElement('jp-data-grid-cell') as DataGridCell;
       if (item.isMatrix) {
         cell.title = 'View Contents';
         cell.className = 'jp-VarInspector-inspectButton';
         const ico = searchIcon.element();
```
Comment (fcollonval) [inline src/variableinspector.ts:311]: Would you mind using a `jp-button` to wrap the icon in it (you can use the appearance _stealth_)?

File: src/variableinspector.ts [hunk 6]
```diff
@@ -250,48 +291,62 @@ export class VariableInspectorPanel
       }
 
       // Add delete icon and onclick event
-      let cell = row.insertCell(0);
+      let cell = document.createElement('jp-data-grid-cell') as DataGridCell;
       cell.title = 'Delete Variable';
       cell.className = 'jp-VarInspector-deleteButton';
+      cell.gridColumn = '1';
       const ico = closeIcon.element();
```
Comment (fcollonval) [inline src/variableinspector.ts:298]: Could you wrap this icon in a `jp-button` with appearance _stealth_?

File: src/variableinspector.ts [hunk 7]
```diff
@@ -8,12 +8,25 @@ import { DockLayout, Widget } from '@lumino/widgets';
...
+} from '@jupyter/web-components';
+provideJupyterDesignSystem().register(allComponents);
+
 import wildcardMatch from 'wildcard-match';
 
 const TITLE_CLASS = 'jp-VarInspector-title';
 const PANEL_CLASS = 'jp-VarInspector';
 const TABLE_CLASS = 'jp-VarInspector-table';
-const TABLE_BODY_CLASS = 'jp-VarInspector-content';
+// const TABLE_BODY_CLASS = 'jp-VarInspector-content';
```
Comment (fcollonval) [inline src/variableinspector.ts:29]: Dead code?

```suggestion
```

File: src/variableinspector.ts [hunk 8]
```diff
@@ -387,27 +439,27 @@ namespace Private {
...
-    const nameOption = document.createElement('option');
+    const nameOption = document.createElement('jp-option') as Option;
     nameOption.value = 'name';
     nameOption.innerHTML = 'Name';
     filterType.appendChild(varTypeOption);
     filterType.appendChild(nameOption);
     const searchContainer = document.createElement('div');
     searchContainer.className = 'jp-InputGroup filter-search-container';
-    const input = document.createElement('input');
+    const input = document.createElement('jp-search') as Search;
```
Comment (fcollonval) [inline src/variableinspector.ts:455]: The [jp-text-field](https://jupyterlab-contrib.github.io/jupyter-ui-toolkit/?path=/docs/components-text-field--documentation) will be more appropriate than the search component.

**Review by fcollonval — CHANGES_REQUESTED** [review]
Thanks @kentarolim10 

Here are some suggestions. And in addition to those there is one point that needs to be addressed:

Could you please add the following statement in the `variableinspector` plugin (in index.ts):

```ts
addJupyterLabThemeChangeListener();
```

This will link the toolkit theme with the JupyterLab theme.

**Comment by kentarolim10** [comment]
@fcollonval I have made the request changes to the PR 👍

**Review by fcollonval — APPROVED** [review]
Thanks @kentarolim10

**Pull request merged**